src/ConstructiveSolidGeometry/VolumePrimitives/Box.jl

```diff
@@ -1,84 +1,94 @@
+# rotate_aroud_own_center(b::Box{T, CO}, r::AbstractMatrix{T}) where {T, CO} = Box{T, CO}(b.hX, b.hY, b.hZ, b.origin, r * b.rotation)
+(*)(r::AbstractMatrix, b::Box) = rotate(b, r)
 
-in(p::AbstractCoordinatePoint, b::Box) =
-    _in_x(p, b.x) && _in_y(p, b.y) && _in_z(p, b.z)
-    
-    
-# read-in
-function Geometry(::Type{T}, ::Type{Box}, dict::AbstractDict, input_units::NamedTuple) where {T}
+_in(pt::CartesianPoint, b::Box{<:Any, ClosedPrimitive}) =
+    abs(pt.x) <= b.hX && abs(pt.y) <= b.hY && abs(pt.z) <= b.hZ
+_in(pt::CartesianPoint, b::Box{<:Any, :OpenPrimitive}) = 
+    abs(pt.x) < b.hX && abs(pt.y) < b.hY && abs(pt.z) < b.hZ
+ 
+function Geometry(::Type{T}, ::Type{Box}, dict::AbstractDict, input_units::NamedTuple, transformations) where {T}
     length_unit = input_units.length
     x = parse_interval_of_primitive(T, "x", dict, length_unit)
     y = parse_interval_of_primitive(T, "y", dict, length_unit)
     z = parse_interval_of_primitive(T, "z", dict, length_unit)
-    return Box(T, x, y, z)
+    μx = typeof(x) <: Real ? zero(T) : mean(x)
+    μy = typeof(y) <: Real ? zero(T) : mean(y)
+    μz = typeof(z) <: Real ? zero(T) : mean(z)
+    origin = CartesianPoint{T}(μx, μy, μz)
+    scale = ones(SVector{3,T})
+    hX = typeof(x) <: Real ? x : width(x)/2
+    hY = typeof(y) <: Real ? y : width(y)/2
+    hZ = typeof(z) <: Real ? z : width(z)/2
+    box = Box{T, ClosedPrimitive}(
+        scale[1] * hX, 
+        scale[2] * hY, 
+        scale[3] * hZ, 
+        origin
+    )
+    transform(box, transformations)
 end
 
-function Dictionary(b::Box{T}) where {T}
-    dict = OrderedDict{String,Any}()
-    dict["x"] = typeof(b.x) == T ? b.x * 2 : OrderedDict{String,Any}("from" => b.x.left, "to" => b.x.right)
-    dict["y"] = typeof(b.y) == T ? b.y * 2 : OrderedDict{String,Any}("from" => b.y.left, "to" => b.y.right)
-    dict["z"] = typeof(b.z) == T ? b.z * 2 : OrderedDict{String,Any}("from" => b.z.left, "to" => b.z.right)
-    OrderedDict{String,Any}("box" => dict)
+function vertices(b::Box{T}) where {T}
+    return SVector{8, CartesianPoint{T}}(
+        b.rotation * SVector{3,T}(-b.hX, -b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, -b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, +b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, +b.hY, -b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, -b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, -b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(+b.hX, +b.hY, +b.hZ) .+ b.origin,
+        b.rotation * SVector{3,T}(-b.hX, +b.hY, +b.hZ) .+ b.origin,
+    )
 end
 
-get_x_limits(b::Box{T}) where {T} = (_left_linear_interval(b.x), _right_linear_interval(b.x))
-get_y_limits(b::Box{T}) where {T} = (_left_linear_interval(b.y), _right_linear_interval(b.y))
-get_z_limits(b::Box{T}) where {T} = (_left_linear_interval(b.z), _right_linear_interval(b.z))
+sample(b::Box) = vertices(b)
```

Comment:
Will this be the only sample method? If it is what will sample look like in a torus or cone?

Reply:
We only need the sample method to get important points for the initialization of the grid. 
For the box this works right now. Though we might want to extend it to `sample(b::Box, n::NTuple{3, Int} = (2,2,2))`.
Same for all other primitives. E.g. `sample(cone::Box, n::NTuple{3, Int} = (2,2,2))`. 
(`n` could be made configurable in the configuration file. So the user can influence the initialization of the grid.)

Reply:
Because otherwise, we will use the new painting algorithm that does not care about `sample`?